Query count failure in users.tests.test_api.UserTestCase

### NetBox Edition

NetBox Community

### NetBox Version

main branch

### Python Version

3.14

### Steps to Reproduce

1. Fresh checkout of main branch
2. Empty database
3. run `NETBOX_CONFIGURATION=netbox.configuration_testing ./manage.py test users.tests.test_api.UserTestCase`

### Expected Behavior

Tests pass.
I noticed this error first in the pipeline in netbox-docker. But can reproduce it with a non-docker installation from the current main branch.
https://github.com/netbox-community/netbox-docker/actions/runs/26615203017/job/78429277508?pr=1694
https://github.com/netbox-community/netbox-docker/actions/runs/26594617626/job/78362114026?pr=1692

### Observed Behavior

```
ETBOX_CONFIGURATION=netbox.configuration_testing ./manage.py test users.tests.test_api.UserTestCase
Found 24 test(s).
Using existing test database for alias 'default'...
System check identified no issues (0 silenced).
.............F........s.
======================================================================
FAIL: test_list_objects (users.tests.test_api.UserTestCase.test_list_objects)
GET a list of objects as an authenticated user with permission to view the objects.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tgenannt/LocalGit/netbox/netbox/utilities/testing/api.py", line 197, in test_list_objects
    with assert_expected_query_count(self, 'api_list_objects'):
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tgenannt/.local/share/uv/python/cpython-3.14.4-linux-x86_64-gnu/lib/python3.14/contextlib.py", line 148, in __exit__
    next(self.gen)
    ~~~~^^^^^^^^^^
  File "/home/tgenannt/LocalGit/netbox/netbox/utilities/testing/query_counts.py", line 137, in assert_expected_query_count
    test_case.fail(
    ~~~~~~~~~~~~~~^
        f"Query count for {app_label}/{key} changed: "
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
        f"to update the baseline.\nObserved queries:\n{sample}"
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
AssertionError: Query count for users/user:api_list_objects changed: expected 12, got 11. If this change is intentional, re-run with UPDATE_QUERY_COUNTS=1 to update the baseline.
Observed queries:
  1. SELECT "users_token"."id", "users_token"."version", "users_token"."user_id", "users_token"."description", "users_token"."created", "users_token"."expires", "users_token"."last_used", "users_token"."enabled", "users_token"."write_enabled", "
  2. SELECT "users_user"."id", "users_user"."password", "users_user"."last_login", "users_user"."is_superuser", "users_user"."username", "users_user"."first_name", "users_user"."last_name", "users_user"."email", "users_user"."is_active", "users_
  3. UPDATE "users_token" SET "last_used" = '2026-05-29 06:09:57.379138+00:00'::timestamptz WHERE "users_token"."id" = 38
  4. SELECT DISTINCT ON ("users_objectpermission"."id") "users_objectpermission"."id", "users_objectpermission"."name", "users_objectpermission"."description", "users_objectpermission"."enabled", "users_objectpermission"."actions", "users_object
  5. SELECT ("users_objectpermission_object_types"."objectpermission_id") AS "_prefetch_related_val_objectpermission_id", "django_content_type"."id", "django_content_type"."app_label", "django_content_type"."model" FROM "django_content_type" INN
  6. SELECT "django_content_type"."app_label" AS "content_type__app_label", "auth_permission"."codename" AS "codename" FROM "auth_permission" INNER JOIN "users_user_user_permissions" ON ("auth_permission"."id" = "users_user_user_permissions"."pe
  7. SELECT "django_content_type"."app_label" AS "content_type__app_label", "auth_permission"."codename" AS "codename" FROM "auth_permission" INNER JOIN "users_group_permissions" ON ("auth_permission"."id" = "users_group_permissions"."permission
  8. SELECT COUNT(*) AS "__count" FROM "users_user" WHERE "users_user"."id" IN (SELECT U0."id" FROM "users_user" U0 WHERE U0."id" IN (34, 35))
  9. SELECT "users_user"."id", "users_user"."password", "users_user"."last_login", "users_user"."is_superuser", "users_user"."username", "users_user"."first_name", "users_user"."last_name", "users_user"."email", "users_user"."is_active", "users_
  10. SELECT ("users_user_groups"."user_id") AS "_prefetch_related_val_user_id", "users_group"."id", "users_group"."name", "users_group"."description" FROM "users_group" INNER JOIN "users_user_groups" ON ("users_group"."id" = "users_user_groups".
  11. SELECT ("users_user_object_permissions"."user_id") AS "_prefetch_related_val_user_id", "users_objectpermission"."id", "users_objectpermission"."name", "users_objectpermission"."description", "users_objectpermission"."enabled", "users_object

----------------------------------------------------------------------
Ran 24 tests in 6.414s

FAILED (failures=1, skipped=1)
```

--- a/netbox/users/tests/test_api.py
+++ b/netbox/users/tests/test_api.py
@@ -39,25 +39,25 @@ class UserTestCase(APIViewTestCases.APIViewTestCase):
         permissions[2].object_types.add(ObjectType.objects.get_by_natural_key('dcim', 'rack'))
 
         users = (
-            User(username='User1', password='FooBarFooBar1'),
-            User(username='User2', password='FooBarFooBar2'),
-            User(username='User3', password='FooBarFooBar3'),
+            User(username='user1', password='FooBarFooBar1'),
+            User(username='user2', password='FooBarFooBar2'),
+            User(username='user3', password='FooBarFooBar3'),
         )
         User.objects.bulk_create(users)
 
         cls.create_data = [
             {
-                'username': 'User4',
+                'username': 'user4',
                 'password': 'FooBarFooBar4',
                 'permissions': [permissions[0].pk],
             },
             {
-                'username': 'User5',
+                'username': 'user5',
                 'password': 'FooBarFooBar5',
                 'permissions': [permissions[1].pk],
             },
             {
-                'username': 'User6',
+                'username': 'user6',
                 'password': 'FooBarFooBar6',
                 'permissions': [permissions[2].pk],
             },

